### PR TITLE
Support BLS

### PR DESCRIPTION
**Pull request by xgreenx**
Integrated blst implementation of BLS with DST from FileCoin.
It contains a simple functions:
Sign - Sign some data with BLSPrivateKey
Verify - Verify that data with signature is signed by a public key
AggregateSignatures - Aggregate several signatures to one signature
AggregatePublicKeys - Aggregate several public keys to one public key
AggregateVerify - Verify that each data was signed by according pub key and it equals to aggregated signature
FastAggregateVerify - Verify that message is signed by all public keys

According change ob blockchain side: https://github.com/proximax-storage/cpp-xpx-chain/pull/271

### CODE REVIEW

File: sdk/bls.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+var ZeroBLSPublicKey = BLSPublicKey(make([]byte, 48))
+
+// It must be a 96 byte sequence.
+type BLSSignature string
+
+var ZeroBLSSignature = BLSSignature(make([]byte, 96))
+
+func GeneratePrivateKey(seed io.Reader) BLSPrivateKey {
+	if seed == nil {
```
Comment (jenyasd209) [inline sdk/bls.go:31]: just want to clarify if we should handle `nil` interfaces that actually are not `nil`? For example, if I make something like this
```
var zr *ZeroReader
GeneratePrivateKey(zr)
```

i'll catch nil pointer panic. In this case, `zr` is `nil`, but `seed` actually is not `nil`

File: sdk/bls.go [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+	if len(keys) == 0 {
+		return ZeroBLSPublicKey, nil
+	}
+	temp := make([][]byte, len(keys))
+	aggregator := new(blst.P1Aggregate)
+	for i, key := range keys {
+		temp[i] = []byte(key)
+	}
+	if !aggregator.AggregateCompressed(temp, true) {
+		return ZeroBLSPublicKey, errors.New("unable to aggregate signature")
```
Comment (jenyasd209) [inline sdk/bls.go:131]: Are you sure that `AggregatePublicKeys` should return the error `unable to aggregate signature`?:D

**Review by jenyasd209 — APPROVED** [review]

File: sdk/bls.go [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+	"io"
+	"reflect"
+	"strings"
+)
+
+const FILECOIN_DST = string("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_")
+const ETH2_DST = string("BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_")
+
+// It must be a 32 byte sequence.
+type BLSPrivateKey string
```
Comment (Wondertan) [inline sdk/bls.go:17]: Why string and not []byte?

**Review by Wondertan — APPROVED** [review]

File: sdk/bls.go (hunk 3 above)
Comment (xgreenx) [reply sdk/bls.go:17]: Because I can't create const byte[]))

**Pull request merged**